src/enumerable.cr

```diff
@@ -198,6 +198,30 @@ module Enumerable(T)
     n.times { each { |x| yield x } }
   end
 
+  # Traverses the depth of a structure and returns the value.
+  # Returns nil if not found.
+  #
+  # ```
+  # h = {"a" => {"b" => [10, 20, 30]}}
+  # h.dig "a", "b", 1             # => 20
+  # h.dig "a", "b", "c", "d", "e" # => nil
+  # ```
+  def dig(key : K, *subkeys)
```

Comment:
Pardon if I just don't get the current state of things, but where is K defined in Enumerable? I thought that type placeholder names which weren't defined in context needed to be declared with forall clauses.